transport-native-epoll/src/main/java/io/netty/channel/epoll/LinuxSocket.java

```diff
@@ -140,6 +152,18 @@ PeerCredentials getPeerCredentials() throws IOException {
         return getPeerCredentials(intValue());
     }
 
+    long sendFile(DefaultFileRegion src, long baseOffset, long offset, long length) throws IOException {
+        // Open the file-region as it may be created via the lazy constructor. This is needed as we directly access
+        // the FileChannel field via JNI.
+        src.open();
+
+        long res = sendFile(intValue(), src, baseOffset, offset, length);
+        if (res >= 0) {
+            return res;
+        }
+        return ioResult("sendfile", (int) res, SENDFILE_CONNECTION_RESET_EXCEPTION, SENDFILE_CLOSED_CHANNEL_EXCEPTION);
```

Comment:
When we interact with JNI we use the pattern of putting `-errno` in the return value if an error occurs. This is so that we don't have to make another JNI call to get the actual error. So this shouldn't be an overflow scenario.